### PR TITLE
Added Show/Hide Menu

### PR DESCRIPTION
**Pull request by ykr5184**
I added a way for people to hide the difficulty and/or problem type tags, in case they want to avoid seeing this for problems they are still attempting or haven't attempted yet.

### CODE REVIEW

File: components/Problemset/ShowMenu/index.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+          >
+            
+            {"Problem Difficulty"}
+          </MenuItemOption>
+          <MenuItemOption 
+            type="checkbox"
+            value={show.showTags ? "tags" : "hidden"}
+            onClick={() => setShow(({ showDiff, showTags }) => ({ showDiff: showDiff, showTags: !showTags }))}
+          >
+            {"Problem Type"}
```
Comment (myst-6) [inline components/Problemset/ShowMenu/index.tsx:56]: Is there a reason this says "Problem Type" and not "Problem Tags"? I think tags makes more sense

**Review by myst-6 — CHANGES_REQUESTED** [review]
I think it's really good. I don't see any problems with it. Let's get a second opinion from @macaquedev as well.

File: components/Problemset/ShowMenu/index.tsx (hunk 1 above)
Comment (vladimirfilip) [reply components/Problemset/ShowMenu/index.tsx:56]: I agree with @myst-6

**Review by vladimirfilip — CHANGES_REQUESTED** [review]
Very useful feature!

There's now a lot of buttons to the right of 'Problems'. I think it would be nice if they were in their own horizontal stack below the heading rather than next to it.

File: components/Problemset/ShowMenu/index.tsx (hunk 1 above)
Comment (macaquedev) [reply components/Problemset/ShowMenu/index.tsx:56]: I also agree with @myst-6

**Review by vladimirfilip — APPROVED** [review]
Perfect

**Review by macaquedev — APPROVED** [review]
Good work!

**Review by myst-6 — APPROVED** [review]
orz

**Pull request merged**